[CS-3088] Font weight work-around

oidoug opened the pull request:
<!-- This is a pull request. Please make sure any applicable bullet points have been covered. -->

### Description

This PR adds supported font-weight variations for font OpenSans-Regular on android.

Differently than on iOS, on Android, the only font used after Theme translation by `restyle` is `OpenSans-Regular`. This can be easily tested by trying to use the wrong font-weight on any Text, it will show an error with the available options for the font.

- [x] Completes #(CS-3088)

### Checklist

- [x] All UI changes have been tested on a small device

### Screenshots

<img width="300" alt="Screenshot" src="https://user-images.githubusercontent.com/129619/154964398-d5352da2-100a-4e97-a925-ca57b741ef82.jpeg">

Commit:
Adding supported font weight variations for font OpenSans-Regular on android.

danibonilha reviewed (commented):
The change looks good, but I noticed there are places we are using `fontWeight` and other `weight`,  in order to to apply this across the app we would need to replace the `fontWeight` with `weight` in all used Text reestyle components and not allow `Text` to receive `fontWeight` anymore to avoid making a android mistake, an easy way, is to apply this on the `Text.ts` line 13: 

```ts
export type TextProps = Omit<ShopifyTextProps<Theme>, 'fontWeight'> &
```

Commit:
Refactor all uses of fontWeight to Theme defined 'weight' prop in Text component.

Commit:
Merge branch 'develop' into doug/font-weight-android

danibonilha commented on src/components/walletconnect-list/WalletConnectExplainer.js:
```diff
@@ -25,10 +25,10 @@ const IconQRScanImg = styled(ImgixImage).attrs({
     size="xs"
     textAlign="center"
+    weight={props.bold ? 'bold' : 'regular'}
     {...props}
```
nit: This one already passes the props forward, we could remove this condition, and on `   <CustomText bold>` replace with `   <CustomText weight='bold'>`

oidoug replied:
Good catch 👍

danibonilha reviewed (commented):
small suggestion, and missing some eslint fixes, once it's green, it's good to go.

Commit:
Fixes order of props ts-error

Commit:
Refactors unneded prop on CustomText

oidoug commented:
@danibonilha , I refactored all Texts to use weight and added the Omit as you suggested.

There are a few places that are still using `fontWeight` but they are either for Inputs or are not using the Theme (prob. older Rainbow implementation). If you don't mind I'll leave these cases behind for now.

danibonilha reviewed (approved):
🚀

Commit:
Merge branch 'develop' into doug/font-weight-android
